packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -159,7 +160,10 @@ export class Pipeline extends cdk.Construct implements IPipeline {
     // If a bucket has been provided, use it - otherwise, create a bucket.
     let propsBucket = props.artifactBucket;
     if (!propsBucket) {
+      const encryptionKey = new kms.EncryptionKey(this, 'ArtifactsBucketEncryptionKey');
```

Comment:
Feels like something we should allow users to opt out from and also supply their own key.

Reply:
We already allow it - the customer can supply their own `artifactBucket` in `props` (notice that we check `if (!propsBucket)`, and don't do any defaulting in that case).

Reply:
My comment was about the key, not about the bucket, but I guess you are right. If they want to customize the key, they can just pass in a custom bucket. Make sure to mention in the `artifactBucket` props that the bucket is encrypted.